### DISCUSSION TITLE
[Node] supports Security S0, but the S0 network key was not configured. Th e interview might not include all functionality.

### DISCUSSION THREAD

**Discussion by karmingc** [zwave-js/zwave-js#7517] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?

We are currently trying to pair a Yale lock (Assure Touchscreen Deadbolt YRD226) and seem to missing some entities in Home Assistant 
that we usually expect. We tried excluding and including multiple times, as well as performing a factory reset on the door lock and we still seeing the following error:

```
[Node] supports Security S0, but the S0 network key was not configured. Th e interview might not include all functionality.
```

We are currently using zwave-js-ui and those fields are populated.

What did you expect to happen instead?

The S0 network key error should be present.

We have multiple raspberry pis but this seems to be happening for a few devices, despite the fact they all have the same configuration file. Reproducing is hard.

Anything else we should know?

### Software versions

Z-Wave JS UI: https://github.com/zwave-js/zwave-js-ui/releases/tag/v9.22.0

Home Assistant Z-Wave Integration: 2024.9.2

### Z-Wave Controller (Stick/Dongle/...)

discussions

### Device information

AEON Labs Z-Stick Gen5 USB Controller ZW090
YALE Assure Touchscreen Deadbolt

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwavejs_2024-12-23.log](https://github.com/user-attachments/files/18232115/zwavejs_2024-12-23.log)

**Comment by AlCalzone** [comment, marked as answer]
Update Z-Wave JS UI. There were some intermediate versions after releasing support for secondary controllers which would not use the configured security keys, because I assumed you'd only have a controller with the secondary role after joining a network.

Turns out there are a bunch of folks out there running their networks with a controller that's not a primary controller.

**Reply by karmingc** [reply]
Thanks @AlCalzone, I can confirm this worked (forgot to update this thread). I updated to 9.29 and it worked onwards!
